lib/seahorn/Smt/Yices2SolverImpl.cc

```diff
@@ -27,16 +27,24 @@ inline void yices_library_initialize(void){
   }
 }
 
-yices_solver_impl::yices_solver_impl(expr::ExprFactory &efac, solver_options opts)
-  : Solver(),
-    d_efac(efac) {
-  
+yices_solver_impl::yices_solver_impl(expr::ExprFactory &efac, const char *logic,
+                                     solver_options opts)
+    : Solver(), d_efac(efac) {
+
   yices_library_initialize();
   /* the yices configuration structure */
-  ctx_config_t *cfg = nullptr;
-  
+  ctx_config_t *cfg = yices_new_config();
+  if (logic != nullptr) {
+    int32_t res = yices_default_config_for_logic(cfg, logic);
+    if (res) {
+      errs() << "Warning: logic type [" << logic
+             << "] not supported by Yices2;"
+                "Falling back to default(QF_AUFBV). \n ";
```

Comment:
maybe just don't set logic. Yices2 seems to work fine without explicit logic command. Just report and continue.